### PR TITLE
LTS 11, squashed

### PR DESCRIPTION
**Pull request by neongreen**
Git is complicated.

### CODE REVIEW

**Comment by neongreen** [comment]
Don't merge even with Matthias's approval, I will merge. You won't merge. You won't merge!

File: libs/types-common-journal/types-common-journal.cabal [hunk 1]
```diff
@@ -25,7 +25,9 @@ library
         Data.Proto
         Data.Proto.Id
         Proto.TeamEvents
+        Proto.TeamEvents_Fields
         Proto.UserEvents
+        Proto.UserEvents_Fields
```
[the remark this answers is not in the document]
Comment (tiago-loureiro) [reply libs/types-common-journal/types-common-journal.cabal]: This is autogenerated code, which is why there are no changes in `UserEvents`

File: libs/ssl-util/src/Ssl/Util.hs [hunk 2]
```diff
@@ -200,32 +162,31 @@ verifyRsaFingerprint d = verifyFingerprint $ \pk ->
...
+--
+-- Throws an error for other types of connections.
+withVerifiedSslConnection
+    :: (SSL -> IO ())       -- ^ A function to verify fingerprints given an SSL connection
+    -> Manager
+    -> Request              -- ^ Request (needed to open a new connection if
+                            -- there isn't one available yet). Take care to
+                            -- use the request passed to the callback
+                            -- instead of the one passed to
+                            -- 'withVerifiedSslConnection'
```
[the remark this answers is not in the document]
Comment (neongreen) [reply libs/ssl-util/src/Ssl/Util.hs]: This comment means "You give us request you want to execute, we modify it magically and give it back. Please use the modified request and not the original one."

I changed the code to make it harder to use the wrong request.

[the remark this answers is not in the document]
Comment (neongreen) [reply libs/ssl-util/src/Ssl/Util.hs]: (Will ping when the changes are pushed.)

File: services/brig/src/Brig/App.hs [hunk 3]
```diff
@@ -315,13 +315,22 @@ initHttpManager = do
...
+-- Note [SSL context]
+-- ~~~~~~~~~~~~
+--
+-- 'openSslManagerSettings' takes an IO action for creating the context;
+-- presumably a new context is created for each connection, then. However,
+-- judging by comments at https://github.com/snoyberg/http-client/pull/227,
+-- it should be fine to reuse the context, and reusing it is probably
+-- faster. So, we reuse the context.
+
+initExtGetManager :: IO (Manager, [Fingerprint Rsa] -> SSL.SSL -> IO ())
```
[the remark this answers is not in the document]
Comment (neongreen) [reply services/brig/src/Brig/App.hs]: I think it's Brig-specific – we have some custom SSL logic that is only used for Brig<->bots communication.

[the remark this answers is not in the document]
Comment (neongreen) [reply services/brig/src/Brig/App.hs]: Oh well, and also Galley, right. I forgot :/

Yeah, then I'll add a comment about duplication.

**Pull request merged**